### ISSUE TITLE
Types are sometimes both "OK to update" and "require alias"

### ISSUE THREAD

**Issue by runarorama** [unisonweb/unison#990]
```
 I found and typechecked these definitions in ~/work/unison/scratch.u. If you do
  an `add` or `update`, here's how your codebase would change:
    ⍟ These new definitions will replace existing ones of the same name and are ok
      to `update`:
      unique type io.Mode  (existing name: .builtin.io.Mode)
    x These definitions would fail on `add` or `update`:
      Reason
      needs alias   unique type io.Mode   (existing name: .builtin.io.Mode)
      Tip: Use `help filestatus` to learn more.
  Now evaluating any watch expressions (lines starting with `>`)...
  Ctrl+C cancels.
```

This is with a codebase where `.builtin.io.Mode` is the exact same `unique` type (same seed) as the definition in my scratch file.

**Comment by runarorama** [comment]
Doing an `update` at this point actually breaks the codebase:

```
.base> view io.IO

  -- The name io.IO is assigned to the reference #jp3u8, which is missing from the
  codebase.
  Tip: You might need to repair the codebase manually.
```

**Comment by runarorama** [comment]
Fixed by #1254
